cachi2/core/package_managers/pip.py

```diff
@@ -1795,17 +1808,21 @@ def _process_package_distributions(
             docs=PIP_NO_SDIST_DOC,
         )
 
-    best_sdist = max(sdists, key=_sdist_preference)
-    if best_sdist.is_yanked:
-        raise PackageRejected(
-            f"All sdists for package {name}=={version} are yanked",
-            solution=(
-                f"Please update the {name} version in your requirements file.\n"
-                "Usually, when a version gets yanked from PyPI, there will already "
-                "be a fixed version available.\n"
-                "Otherwise, you may need to pin to the previous version."
-            ),
-        )
+    if len(sdists) == 0 and len(wheels) > 0:
+        log.warning("No sdists found for package %s==%s", name, version)
+        best_sdist = None
+    else:
+        best_sdist = max(sdists, key=_sdist_preference)
+        if best_sdist.is_yanked:
+            raise PackageRejected(
+                f"All sdists for package {name}=={version} are yanked",
```

Comment:
Hmm, being yanked is another thing we handle for sdists but not for wheels.

It might be hard to come up with any sane way to handle yanked-ness in an allow_binary context. Maybe we should just log a warning and download yanked things anyway (for consistency: even when allow_binary=False). That's how 'yanked' is supposed to work when the version is pinned, after all. https://peps.python.org/pep-0592/#installers

Reply:
But that's probably something for a follow-up story

Reply:
So, in the scenario that a package has wheels but all sdists are yanked, the request would still fail, right?

Should we file the follow-up story right away so we don't forget about it?